apiutils/task.go

```diff
@@ -0,0 +1,21 @@
+package apiutils
```

Comment:
Agent and dispatcher should use the exact logic to decide if a task update should trigger update or not. Because this is `a very specific kind of equality`, I don't want to copy. Any update to one but not the other may result in hard to find bug.